packages/ui-extensions-server-kit/src/ExtensionServerClient/ExtensionServerClient.ts

```diff
@@ -160,12 +162,21 @@ export class ExtensionServerClient implements ExtensionServer.Client {
 }
 
 function mergeOptions(options: ExtensionServer.Options, newOptions: ExtensionServer.Options) {
-  return {
+  return getValidatedOptions({
     ...options,
     ...newOptions,
     connection: {
       ...options.connection,
       ...newOptions.connection,
     },
-  };
+  });
+}
+
+function getValidatedOptions<TOptions extends DeepPartial<ExtensionServer.Options>>(
+  options: TOptions,
+): TOptions {
```

Comment:
I think this should return `ExtensionServer.Options`, no?

Reply:
oops 🙃

Reply:
This is not possible until the issue with `DeepPartial` is fixed - it complains about setting `protocols`